src/Rules/Rule.cs

```diff
@@ -51,9 +51,14 @@ private RuleInfo GetRuleInfoFromAttributes()
             return null;
         }
 
-        public abstract EvaluationCode Evaluate(IA11yElement element);
+        public virtual EvaluationCode Evaluate(IA11yElement element)
+        {
+            // This base class function should never be called
+            // once all rules have been converted to use RuleInfo.EvaluationCode, this function will be removed
+            throw new NotImplementedException();
+        }
 
-        public bool PassesTest(IA11yElement element)
+        public virtual bool PassesTest(IA11yElement element)
         {
             // This base class function should never be called
             // once all rules have been converted to use RuleInfo.EvaluationCode, this function will be designated abstract
```

Comment:
It seems like this would follow the same pattern as the Evaluate method and be removed once the migration is complete. If that's true, then the comment is incorrect. Maybe:
```suggestion
            // once all rules have been converted to use RuleInfo.EvaluationCode, this function will be removed
```